File: julia/hamming/hamming.jl
function distance(s1::AbstractString, s2::AbstractString)
  distance = 0
  if length(s1) != length(s2) 
    throw(ArgumentError("Strings must be same length"))
  end

  for (index, value) in enumerate(s1)
    if value != s2[index]
      distance += 1
    end
  end
  distance
end

File: go/hamming/hamming.go
package hamming

import (
	"errors"
	"fmt"
)

const testVersion = 5

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("Strings must be the same length")
	}
	l := len(a)
	d := 0
	for i := 0; i < l; i++ {
		if a[i] != b[i] {
			d++
		}
	}
	return d, nil
}
